lib/dependabot/update_checkers/java_script/npm_and_yarn/version_resolver.rb

```diff
@@ -188,6 +234,25 @@ def satisfies_peer_reqs_on_dep?(version)
             end
           end
 
+          def latest_version_of_dep_with_satisfied_peer_reqs(dep)
+            latest_version_finder(dep).
+              possible_versions_with_details.
+              find do |version, details|
+                next false unless version > version_class.new(dep.version)
+
+                details["peerDependencies"].all? do |peer_dep_name, req|
+                  # Can't handle multiple peer dependencies
+                  next false unless peer_dep_name == dependency.name
```

Comment:
TODO: Might want to check that peed dependencies *from* this new version to other dependencies are still satisfied. `react-apollo` case.